src/client/pages/Geo/GeoMap/hooks/useFetchMosaicLayer.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from 'react'
+
+import { MosaicLayerKey } from 'meta/geo/mosaic'
+
+import { useAppDispatch } from 'client/store'
+import {
+  GeoActions,
+  useAppliedMosaicOptions,
+  useMosaicSelected,
+  useMosaicStatus,
+  useMosaicUrl,
+} from 'client/store/ui/geo'
+import { LayerFetchStatus } from 'client/store/ui/geo/stateType'
+import { useCountryIso } from 'client/hooks'
+import { mapController } from 'client/utils'
+
+export const useFetchMosaicLayer = () => {
```

Comment:
In this case, if this hook was in the Mosaic Section, the re fetch would depend on whether or not that component is currently rendered. If the user were to change the country while the `Satellite Mosaic` nav section is not collapsed, then the mosaic layer wouldn't be fetched for the new country because the hook is not present. 
It worked in the previous UI because the logic was in the "Background" button, which was always on the screen.

Reply:
For the country iso change for the other layers we have the hook `useCountryIsoChangeHandler`. I can add some more logic there to also fetch the mosaic layer, and then I can move the rest of this hook to the Mosaic Section component. What do you think?